src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,24 @@
-import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
+
+export const TodoFilter = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
 
-export const TodoFilter: React.FC = () => {
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={event => {
+              dispatch(
+                filterSlice.actions.setStatus(event.target.value as Status),
```

Comment:
Ensure that the `setStatus` action correctly handles the `Status` type. If `event.target.value` is not a valid `Status`, it could lead to runtime errors.